Xây dựng giao diện cho màn 3

namnt520 opened the pull request:
- Thêm thước 
- Thêm thanh trạng thái
- Thêm nút Back
- Thêm hình nền cho màn chơi

namnt520 committed:
Adding file html, css, js to src/ruler

Save the first progress

namnt520 committed:
Add html, css, js file

Error can't render react component

namnt520 committed:
edit html for custormize html csc #1
change it

namnt520 committed:
add progress bar
add back button
add progress bar

namnt520 committed:
add ruler
add ruler to the game_field_under

namnt520 committed:
save before configuring some thing to fix react not run

namnt520 committed:
add background
add background as should be seen like in the dragonlearn

namnt520 committed:
covert box content from html to react jsx in Ruler.js

namnt520 committed:
Merge branch 'testing' into fox

namnt520 committed:
refactor some redundant file, move all root component to Ruler.js
stage #3 should look pretty now, but there's still some event needed to be handled

NIXsnowCat commented on src/ruler/Ruler.js:
```diff
@@ -2,191 +2,197 @@ import React from 'react';
-                        <i id="ball4" className="rotate_ball_4" style={{left: "100px"}}></i>
-                        <i id="ball5" className="rotate_ball_5" style={{left: "124px"}}></i>                            
+        <div class="page">
+            <div class="bg_cell"></div>
```
Ops, quên mất không đổi sang className.
Cơ mà nó vẫn chạy được, và chạy đúng...

umuluto committed:
Thêm 1 dependency: react-draggable

umuluto committed:
Thêm ảnh cho màn 1, thay đổi liên kết trong public/index.html

umuluto committed:
Thêm các thành phần và chức năng kéo thả cơ bản

namnt520 committed:
Thêm quả cầu

Quả cầu có thể di chuyển tự do theo phương ngang
Khi đưa trỏ chuột vào quả cầu, nó sẽ mờ đi
Thêm thanh đối xứng từ quả cầu xuống thước

namnt520 committed:
Thêm hiệu ứng cho quả cầu

Khi di chuyển quả cầu, sẽ có một thanh hiển ra hỗ trợ việc xác định vị trí
Khi quả cầu ở đúng hoặc sai vị trí, màu của nó sẽ được thay đổi

namnt520 committed:
Thêm sự kiện khi kéo quả cầu

Nếu vị trí là đúng, một quả cầu trên thanh trạng thái sẽ được kéo sang phải
Nếu vị trí là sai, quả cầu gần nhất được kéo sang phải sẽ được kéo trở lại vị trí ban đầu
Trong quá trình vẽ thước, do không căn chỉnh đúng, điều này khiến cho việc tính toán đúng sai trong sự kiện kéo thả trở nên sai lệch từ 5-10px
Hiệu ứng chuyển giữa các trạng thái cầu vẫn chưa ổn, cần được điều chỉnh lại

Commit:
Làm lại toàn bộ, vịtrí thành trạng thái toa tàu

close #5, close #7

namnt520 committed:
Thêm màn hình hoàn thành bài học

Sau khi hoàn thành yêu cầu, người chơi được chuyển tới màn hình chúc mừng
Nhấn vào 'Go to lesson' để trở lại màn hình chọn màn chơi

umuluto committed:
Thêm vào thành phần Holder

Holder hiển thị ô chứa toa tàu kèm theo thông báo tùy tham số truyền
vào.

umuluto committed:
Thêm logic và các tương tác kéo thả của màn 1

close #8, close #13

umuluto committed:
Thêm tủy chọn xử lý woff cho file-loader

xem #16

umuluto committed:
Đổi font màn 1 thành PT Sans

close #16

umuluto committed:
Cài đặt lại publicPath của file-loader

umuluto committed:
Build lần đầu

umuluto committed:
Update README.md

umuluto committed:
Đổi react-draggable thành dependency từ devdependency

umuluto committed:
Thêm hoạt ảnh tàu rời ga cho màn 1

close #17

namnt520 committed:
Hoàn thiện các chức năng còn thiếu

Sửa lại css cùng với hàm tính toán vị trí
Chỉnh sửa lại cách chuyển trạng thái của bóng
Căn chỉnh lại màu sắc các thành phần
Thêm điều hướng mũi tên phía trước nút Back
Loại bỏ đi những phần không cần thiết, thêm comment cho các hàm

namnt520 committed:
Merge branch 'master' into fox

umuluto committed:
Thêm vào và cài đặt 1 devDependency: Cypress

close #20

umuluto committed:
Thêm vào bộ kiểm thử cho màn 1

close #21

umuluto committed:
Cài đặt để record bằng Cypress

namnt520 committed:
Merge branch 'master' into fox

umuluto committed:
Delete Types.js

umuluto committed:
Delete DraggableWagon.js

umuluto committed:
Update package.json
